test/server/cards/02_SHD/events/Evacuate.spec.ts

```diff
@@ -0,0 +1,104 @@
+describe('Evacuate', function() {
+    integration(function(contextRef) {
+        it('returning all units cards, no leaders', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['evacuate'],
+                    groundArena: ['pyke-sentinel'],
+                    spaceArena: ['cartel-spacer'],
+                    leader: { card: 'finn#this-is-a-rescue', deployed: false },
+                },
+                player2: {
+                    hand: ['death-star-stormtrooper'],
+                    groundArena: ['wampa'],
+                    spaceArena: ['imperial-interceptor'],
+                    leader: { card: 'grand-moff-tarkin#oversector-governor', deployed: false },
+                }
+            });
+            const { context } = contextRef;
+
+            context.player1.clickCard('evacuate');
+            expect(context.player1.hand).toHaveSize(2);
+            expect(context.player1.hand).toContain(context.pykeSentinel);
+            expect(context.player1.hand).toContain(context.cartelSpacer);
+            expect(context.player1.hand).not.toContain(context.evacuate);
+            expect(context.player1.groundArena).toHaveSize(0);
+
+            expect(context.player2.hand).toHaveSize(3);
+            expect(context.player2.hand).toContain(context.deathStarStormtrooper);
+            expect(context.player2.hand).toContain(context.wampa);
+            expect(context.player2.hand).toContain(context.imperialInterceptor);
+            expect(context.player1.groundArena).toHaveSize(0);
+        });
+
+        it('return all unit cards to hand and discard upgrades', function () {
```

Comment:
To save a lot of duplicate code, you could probably merge test 2 and 3 together or at least reduce the code a bit so you don't have to check so many things. So if you don't need the leader for the test, I wouldn't have him in the test setup.
Also would only have one friendly and one enemy unit in this test case, because you already check that multiple units get bounced with the first test case.

Reply:
collapsed test 3 into test 2, removed a few cards to keep it smaller